lib/gollum-lib/sanitization.rb

```diff
@@ -1,378 +1,52 @@
-     'msrow'=>['href', 'id', 'mathbackground', 'mathcolor'],
-     'mstack'=>['align', 'href', 'id', 'mathbackground', 'mathcolor'],
-     'mstyle'=>
-        ['displaystyle',
-         'scriptlevel',
-         'scriptminsize',
-         'scriptsizemultiplier',
-         'href',
-         'id',
-         'mathbackground',
-         'mathcolor'],
-     'msub'=>['subscriptshift', 'href', 'id', 'mathbackground', 'mathcolor'],
-     'msup'=>['supscriptshift', 'href', 'id', 'mathbackground', 'mathcolor'],
-     'msubsup'=>
-        ['subscriptshift',
-         'supscriptshift',
-         'href',
-         'id',
-         'mathbackground',
-         'mathcolor'],
-     'mtable'=>
-        ['align',
-         'columnalign',
-         'columnlines',
-         'columnspacing',
-         'displaystyle',
-         'frame',
-         'framespacing',
-         'rowalign',
-         'rowlines',
-         'rowspacing',
-         'width',
-         'href',
-         'id',
-         'mathbackground',
-         'mathcolor'],
-     'mtd'=>
-        ['columnalign',
-         'columnspan',
-         'rowalign',
-         'rowspan',
-         'href',
-         'id',
-         'mathbackground',
-         'mathcolor'],
-     'mtext'=>
-        ['dir',
-         'mathbackground',
-         'mathcolor',
-         'mathsize',
-         'mathvariant',
-         'href',
-         'id',
-         'mathbackground',
-         'mathcolor'],
-     'mtr'=>
-        ['columnalign', 'rowalign', 'href', 'id', 'mathbackground', 'mathcolor'],
-     'munder'=>
-        ['accentunder', 'align', 'href', 'id', 'mathbackground', 'mathcolor'],
-     'munderover'=>
-        ['accent',
-         'accentunder',
-         'align',
-         'href',
-         'id',
-         'mathbackground',
-         'mathcolor'],
-     'semantics'=>['href', 'id', 'mathbackground', 'mathcolor']
-    }.freeze
+    # This class method is used in the Tag filter to determine whether a link has an acceptable URI scheme.
+    def self.accepted_protocols
+      @@accepted_protocols
+    end
 
-    # Default whitelisted elements.
-    ELEMENTS   = ([
-      'a', 'abbr', 'acronym', 'address', 'area', 'b', 'big',
-      'blockquote', 'br', 'button', 'caption', 'center', 'cite',
-      'code', 'col', 'colgroup', 'dd', 'del', 'dfn', 'dir', 'div',
-      'dl', 'dt', 'em', 'fieldset', 'font', 'form', 'h1', 'h2', 'h3',
-      'h4', 'h5', 'h6', 'hr', 'i', 'img', 'input', 'ins', 'kbd', 'label',
-      'legend', 'li', 'map', 'mark', 'math', 'menu', 'mfrac', 'mi', 'mn',
-      'mo', 'mrow', 'msqrt', 'msubsup', 'msup', 'mtext', 'ol', 'optgroup',
-      'option', 'p', 'pre', 'q', 's', 'samp', 'select', 'small', 'span',
-      'strike', 'strong', 'sub', 'sup', 'table', 'tbody', 'td', 'textarea',
-      'tfoot', 'th', 'thead', 'tr', 'tt', 'u', 'ul', 'var'
-    ] + MATHML_ELEMENTS).freeze
+    REMOVE_NODES = ['style', 'script']
 
+    SCRUB_REMOVE = Loofah::Scrubber.new do |node|
```

Comment:
To explain: the `:strip` scrubber called in `#clean` removes unwhitelisted tags, but keeps their content. This is similar to Sanitize's behavior. However, we had some custom config in place that told Sanitize to complete remove some tags (i.e. also remove their content). This scrubber implements that behavior.

Reply:
Nice!